lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -1116,7 +1116,6 @@ namespace Js
 
     BOOL JavascriptProxy::IsEnumerable(PropertyId propertyId)
     {
-        Assert(FALSE);
```

Comment:
Not sure about this... Does the assertion means that this is never supposed to be called?
After the change in ForInObjectEnumerator::MoveAndGetNext() this can be called for Proxy objects.

Reply:
Our proxy objects do not have any properties themselves, and should handle pretty much everything with a trap.

In this case, if the proxy has a trap for `getOwnPropertyDescriptor`, should we instead be calling that to determine the enumerability of a given property ID?

e.g.

```js
var o = {a: 1, b: 2}
var p = new Proxy(o, {
  getOwnPropertyDescriptor(target, prop) {
    var d = Object.getOwnPropertyDescriptor(target, prop); 
    if (prop == "b") {
      d.enumerable = false;
    }
    return d;
  }
});
for(var x in p) {
	console.log(x + " in p");
}
```
This will report that `a` is enumerable, but `b` is not, while your change here will report both as enumerable.

Reply:
Which is to say: I think this assertion should remain, because it reminds us that we shouldn't be calling into this function.